### ISSUE TITLE
Implement Zoom-into in webUI

### ISSUE THREAD

**Issue by metas-mk** [metasfresh/metasfresh-webui-frontend#720]
### Type of issue
Feature request
### Current behavior
Currently there is no Zoom-into functionality in WebUI (that thing that in Swing can be used w/ right-mouse "Springe zu" and opens the window and record for the selected ID)
### Expected behavior
Have that functionality or similar in WebUI.

Idea: In detail view have the labels then as links to zoom-into. let's solve that first, can think later about grid-view zoom-into.

belongs to https://github.com/metasfresh/metasfresh-webui-api/issues/348

**Comment by teosarca** [comment]
## Who shall work?

1. If a field supports ZoomInto backend provides "supportZoomInto:true" on layout field level.
e.g.
![image](https://cloud.githubusercontent.com/assets/1244701/25770399/fcd7f420-323c-11e7-9100-b5ac3041ac67.png)

2. To zoom into that field, you need to call one of the following endpoints
* field on header document: http://w101.metasfresh.com:8081/swagger-ui.html#!/window-rest-controller/getDocumentFieldZoomIntoUsingGET
* field on included document: http://w101.metasfresh.com:8081/swagger-ui.html#!/window-rest-controller/getDocumentFieldZoomIntoUsingGET_1

The endpoint will return something like:
![image](https://cloud.githubusercontent.com/assets/1244701/25770416/705615c6-323d-11e7-8543-17903dd698ce.png)

Basically the result has 2 properties:
* "documentPath": which contains windowId, documentId of the the document which we need to open
* "windowLayout" : backend automatically prefetched the window layout of the window we are zooming into. It was prefetched just to avoid another backend call. Might be that we don't need this...

## Why not providing the ZoomInfo on each field, when we fetch the document data?
The main reason is because we would need extra computation to figure out which is the actual windowId where we need to zoom.

**Comment by damianprzygodzki** [comment]
How do we want achieve a context menu over widgets? Most of them are editable inputs so the replacing native context menu would be limiting default functionalities. @cadavre We need your help here.

**Comment by cadavre** [comment]
I think going into context menu "Zoom in" is not a good follow-up of Swing's client. ;)

This is how I see we can do it:
If `supportZoomInto:true` – show additional perpendicular (vertical) icon-button next to (on right) of the widget. If now only single option is available ("zoom in") – let this icon-button instantly "zoom into" document selected in widget; if - in the future - there would be more options than just one – we can make icon-button with three dots which will open dropdown with more options.

**and**

As an addition to above I would like to add something like [ctrl]+click widget to "zoom in". Possible @damianprzygodzki ?

**Comment by damianprzygodzki** [comment]
@cadavre yes, it is possible. But i am thinking - you should be focused inside widget then, shouldnt we be consistent and proceed like with devices section (tab navigation)?

**Comment by cadavre** [comment]
I guess it should be possible even without focus. Imagine when someone is browsing document – already filled and want to go to Business Partner to check other documents. Simple go with mouse over widget and click to open. Also remember about case with widget read-only – cannot be focused, but still we want to have possibility to zoom-in.

**Comment by damianprzygodzki** [comment]
Yes, sorry, you are right. We can proceed this way.

**Comment by teosarca** [comment]
Hi guys,

Have you also considered?
> Idea: In detail view have the labels then as links to zoom-into. let's solve that first, can think later about grid-view zoom-into.


I am really not sure if that's a good idea, but i would really want to have your input.

**Comment by cadavre** [comment]
Labels are also not a bad idea. What I would do is add some small "go into record" icon next to label that is enabled to "zoom in". Wdyt?

**Comment by teosarca** [comment]
or in case the widget supports zoomInto, we could render the label as a browser link.

To be honest i am really not sure and I think we would not be able to decide now.
My suggestion is to render it as you think it would look good and which approach is the fastest one and we can evolve from there.
wdyt?

**Comment by cadavre** [comment]
I would not really change label style to link. But afaik I think approach with labels would be fastest. @damianprzygodzki ?

**Comment by damianprzygodzki** [comment]
@cadavre 
> show additional perpendicular (vertical) icon-button next to (on right) of the widget.

And what is the difference between `labels` and `icon-button`? I think in both cases it is the same as Devices `button`.

**Comment by cadavre** [comment]
![untitled](https://cloud.githubusercontent.com/assets/461796/25839916/01b1ce1c-349a-11e7-9923-860954aa7bad.png)

Both ideas on screenshot here.

**Comment by damianprzygodzki** [comment]
Both ideas are ok, but i would go with icon-button to avoid problems with labels on small resolutions.

**Comment by teosarca** [comment]
as discussed, let's go with the label approach.
* same style as a regular label (same size, font, color)
* when hovering, it shall be underlined, like standard web links

The document shall be opened in a new browser tab.

**Comment by damianprzygodzki** [comment]
@teosarca We need just document path (windowId, documentId). Could we just provide in layout:
```supportZoomInto: {
   windowId: 143,
   documentId: 100000
}```
And bam. Thats it. No additional calls.

**Comment by teosarca** [comment]
@damianprzygodzki  

> Why not providing the ZoomInfo on each field, when we fetch the document data?
> The main reason is because we would need extra computation to figure out which is the actual windowId where we need to zoom.

Basically we cannot provide the windowId because it needs computation.
We cannot provide the documentId in layout because the documentId shall be fetch from field value.
The layout is not aware of what data it's inside, it's just static.

**Comment by metasnw** [comment]
testing on w101:

NOK: zoom from sales order to price syste breaks bread crumb navigation:

![image](https://cloud.githubusercontent.com/assets/16079655/25947406/9813f27a-364f-11e7-907f-b5412492b471.png)

**Comment by metasnw** [comment]
http://docs.metasfresh.org/webui_collection/EN/Jumpto.html

**Comment by teosarca** [comment]
Quick tested
* selected a manufacturing order: https://w101.metasfresh.com:8443/window/53009/1000001
* click on "Priority" label
* backend throws an exception but frontend shows no error => it shall display the error

**Comment by metas-lc** [comment]
IT

1) zoom in to a window where the role doesn't have permission:
- go to Print Format window (240), press on advanced edit
- zoom into Jasper Process
=> no window read permission OK

2) Priority label in Manufacturing
- go to Manufactuing Order and try to zoom in to Priority
=> you can't. There is no zoom in on that field OK

3) zoom into price system
- go to sales order and zoom into price system
=> window 146 opens OK??
https://w101.metasfresh.com:8443/window/146/2000837
shouldn't this window open? https://w101.metasfresh.com:8443/window/540320/2000837 

4) try to zoom in to a window when the value is null 
- go to sales order, advanced edit and zoom into Activity
=> error: Cannot fetch ZoomInto infos from a null value OK?

checked several zoom into, from both document view and advance edit and they work fine

**Comment by metas-mk** [comment]
@metas-lc hi lili. No. 4 is OK. For No. 3. Can u please create a follow up task. thx.

**Comment by metas-lc** [comment]
@metas-mk Thank you. Created task for No. 3 https://github.com/metasfresh/metasfresh/issues/1620
